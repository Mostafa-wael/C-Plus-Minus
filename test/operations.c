print("arithmetic");
a = 10;
b = 10;
c = 2;
print(-a == -10); // -10
print(-3-4);
print(a+b == 20); // 20
print(a-b == 0); // 0
print(a/b == 1); // 1
print(a%c == 0); // 0
print(a * b - b + a / b == 91); // 91
print(100-5);
print(-100-5+5);
////////////////////////////////
print("bitwise");
a = 10;
c = 2;
print((a | c)  == 10); // 1010 | 0010 = 1010 = 10
print((a & c)  == 2); // 1010 & 0010 = 0010 = 2
print((a ^ c)  == 8); // 1010 ^ 0010 = 1000 = 8
print((a << c)  == 40); // 1010 << 0010 = 101000 = 40
print((a >> c) == 2); // 1010 >> 0010 = 10 = 2
print((~a) == -11); // -11
print((a<<c)>>c == 10); // 10
////////////////////////////////
print("logic");
a = 1;
b = 0;
print(a); // 1
print(b); // 0
print(a == b); // 0
print(a != b); // 1
print(a < b); // 0
print(a > b); // 1
print(a <= b); // 0
print(a >= b); // 1
////////////////////////////////
print("More logic");
print(!a); // 0
print(a && b); // 0
print(a || b); // 1
print((a && a && b) || a); // 1
exit;