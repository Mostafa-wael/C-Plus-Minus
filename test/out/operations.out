"arithmetic"
Quads() push 10
Quads() pop a

Quads() push 10
Quads() pop b

Quads() push 2
Quads() pop c

2
Quads() push a
Quads() NEG
Quads() push 10
Quads() NEG
Quads() EQ
1
Quads() push 3
Quads() NEG
Quads() push 4
Quads() SUB
-7
Quads() push a
Quads() push b
Quads() ADD
Quads() push 20
Quads() EQ
1
Quads() push a
Quads() push b
Quads() SUB
Quads() push 0
Quads() EQ
1
Quads() push a
Quads() push b
Quads() DIV
Quads() push 1
Quads() EQ
1
Quads() push a
Quads() push c
Quads() MOD
Quads() push 0
Quads() EQ
1
Quads() push a
Quads() push b
Quads() MUL
Quads() push b
Quads() SUB
Quads() push a
Quads() push b
Quads() DIV
Quads() ADD
Quads() push 91
Quads() EQ
1
Quads() push 100
Quads() push 5
Quads() SUB
95
Quads() push 100
Quads() NEG
Quads() push 5
Quads() SUB
Quads() push 5
Quads() ADD
-100
"bitwise"
Quads() push 10
Quads() pop a

Quads() push 2
Quads() pop c

Quads() push a
Quads() push c
Quads() BITWISE_OR
Quads() push 10
Quads() EQ
1
Quads() push a
Quads() push c
Quads() BITWISE_AND
Quads() push 2
Quads() EQ
1
Quads() push a
Quads() push c
Quads() BITWISE_XOR
Quads() push 8
Quads() EQ
1
Quads() push a
Quads() push c
Quads() SHL
Quads() push 40
Quads() EQ
1
Quads() push a
Quads() push c
Quads() SHR
Quads() push 2
Quads() EQ
1
Quads() push a
Quads() COMPLEMENT
Quads() push 11
Quads() NEG
Quads() EQ
1
Quads() push a
Quads() push c
Quads() SHL
Quads() push c
Quads() SHR
Quads() push 10
Quads() EQ
1
"logic"
Quads() push 1
Quads() pop a

Quads() push 0
Quads() pop b

1
0
Quads() push a
Quads() push b
Quads() EQ
0
Quads() push a
Quads() push b
Quads() NEQ
1
Quads() push a
Quads() push b
Quads() LT
0
Quads() push a
Quads() push b
Quads() GT
1
Quads() push a
Quads() push b
Quads() LEQ
0
Quads() push a
Quads() push b
Quads() GEQ
1
"More logic"
Quads() push a
Quads() NOT
0
Quads() push 1
Quads() pop x

Quads() push 0
Quads() pop y

Quads() push x
Quads() push y
Quads() LOGICAL_AND
0
Quads() push x
Quads() push x
Quads() LOGICAL_AND
Quads() push y
Quads() LOGICAL_AND
Quads() push x
Quads() LOGICAL_OR
1
